### PR TITLE
[DNM] Looneytunesdoor

### PR DESCRIPTION
**Pull request by Bucovineanu**
## About The Pull Request

You can now paint windows and walls to look like they're airlocks. Walking into them causes you to be stunned and knocked down for 5 seconds. Probably going to be a black market item.

## Why It's Good For The Game

It's funny and gives way to funny situations.

## Changelog
:cl:
add: Rumor has it that the black market is now offering hacked and upgraded airlock printers which can paint any surface to look like a realistic door with holographic technology.

/:cl:

### CODE REVIEW

File: code/game/objects/items/fakelock_painter.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	. = ..()
+	//var/atom/tgt = target
+	//var/turf/closed/wall/F = target
+	if(!proximity)
+		to_chat(user, span_notice("You need to get closer!"))
+		return
+	if(use_paint(user) && (istype(target, /turf/closed/wall) || istype(target, /obj/structure/window/)))
+		// reads from the airlock painter's `available paintjob` list. lets the player choose a paint option, or cancel painting
+		var/current_paintjob = input(user, "Please select a paintjob for this wall.") as null|anything in sortList(available_paint_jobs)
+		if(!current_paintjob) // if the user clicked cancel on the popup, return
```
Comment (TheChosenEvilOne) [inline code/game/objects/items/fakelock_painter.dm:29]: check distance or move this to attackself.

File: code/game/objects/items/fakelock_painter.dm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			return
+
+		var/airlock_type = available_paint_jobs["[current_paintjob]"] // get the airlock type path associated with the airlock name the user just chose
+		var/obj/machinery/door/airlock/airlock = airlock_type // we need to create a new instance of the airlock and assembly to read vars from them
+		//var/obj/structure/door_assembly/assembly = initial(airlock.assemblytype)
+
+		// applies the user-chosen airlock's icon, overlays and assemblytype to the src airlock
+		use_paint(user)
+		target.icon = initial(airlock.icon)
```
Comment (TheChosenEvilOne) [inline code/game/objects/items/fakelock_painter.dm:38]: I am thinking you could do a dummy object that just hides the wall behind it. I am just not sure if you can bump on it.

File: code/game/objects/items/fakelock_painter.dm (hunk 1 above)
Comment (Bucovineanu) [reply code/game/objects/items/fakelock_painter.dm:29]: I probably should but I got tangled into the overlay bugs to the point of getting tired of this bit of code. Gonna see if I feel like it, I want to do a merge for 'morrows session

File: code/game/objects/items/fakelock_painter.dm (hunk 2 above)
Comment (Bucovineanu) [reply code/game/objects/items/fakelock_painter.dm:38]: If I understood how that's supposed to be done, it'd be neat in making it so I can display airlocks more accurately... overlay bug and all.

**Comment by EdgeLordExe** [comment]
fix it ree

**Pull request closed without merging**